pkg/kafka/partition/committer_refactor.go

```diff
@@ -0,0 +1,136 @@
+package partition
+
+import (
+	"context"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"go.uber.org/atomic"
+)
+
+type refactoredPartitionCommitter struct {
```

Comment:
I'm not a big fan of "refactored..." as a name!

Reply:
I think it will be removed when the old code is deleted? @owen-d